code/modules/clothing/spacesuits/soviet.dm

```diff
@@ -74,22 +77,42 @@
 
 /datum/component/armor_protection_limb/proc/on_equip(obj/item/I, mob/equipper)
 	RegisterSignal(equipper, COMSIG_ATOM_BULLET_ACT, .proc/on_projectile)
+	if(has_overlay)
+		update_overlay()
 
 /datum/component/armor_protection_limb/proc/on_unequip(obj/item/I, mob/unequipper, slot)
 	if(isitem(parent))
 		if(!(I.flags_equip_slot & slotdefine2slotbit(slot)))
 			return
+		if(has_overlay)
+			usr.clear_fullscreen("helmet")
 	UnregisterSignal(unequipper, COMSIG_ATOM_BULLET_ACT)
 
 /datum/component/armor_protection_limb/proc/on_projectile(obj/item/I, var/obj/projectile/proj)
 	if(covered_limbs & GLOB.string_part_flags[proj.def_zone])
 		return adjust_health(max(0, proj.damage - round(proj.distance_travelled * proj.damage_falloff)))
 	return FALSE
 
+/datum/component/armor_protection_limb/proc/update_overlay(mob/user)
+	switch(health)
+		if(91 to INFINITY)
+			return
+		if(61 to 90)
+			usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 1)
+		if(31 to 60)
+			usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 2)
+		if(6 to 30)
+			usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 3)
+		if(-INFINITY to 5)
+			usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 4)
```

Comment:
```suggestion
			user.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 4)
```

Reply:
Switching it to user causes a runtime null.overlay_fullscreen

Reply:
Because it can apparently somehow be used without a user. `usr` and `user` are not interchangeable. `usr` is the reason why it applies the overlay to everyone instead of the specific mob.